app/services/ccms/submitters/obtain_document_id_service.rb

```diff
@@ -37,26 +42,33 @@ def populate_documents
         end
       end
 
-      def request_document_ids
-        submission.submission_documents.each do |document|
-          tx_id = document_id_requestor.transaction_request_id
-          @response = document_id_requestor.call
-          document.ccms_document_id = Parsers::DocumentIdResponseParser.new(tx_id, @response).document_id
-          document.status = :id_obtained
-          document.save!
-          create_history('applicant_ref_obtained', 'document_ids_obtained', xml_request, @response) if submission.save!
-        rescue CcmsError => e
-          document.status = :failed
-          raise CcmsError, e
-        end
+      def request_document_id(document)
+        document_id_requestor = document_id_requestor(document.document_type)
+        response = update_document_and_return_response(document, document_id_requestor)
+        document.save!
+        create_history('applicant_ref_obtained', 'document_ids_obtained', document_id_requestor.formatted_xml, response) if submission.save!
+      rescue CcmsError => e
+        document.status = :failed
+        document.save!
+        create_ccms_failure_history('applicant_ref_obtained', e, document_id_requestor.formatted_xml)
+      end
+
+      def ccms_document_id(dir)
+        Parsers::DocumentIdResponseParser.new(dir.transaction_request_id, dir.call).document_id
+      end
+
+      def document_id_requestor(document_type)
+        Requestors::DocumentIdRequestor.new(submission.case_ccms_reference, submission.legal_aid_application.provider.username, document_type)
       end
 
-      def document_id_requestor
-        @document_id_requestor ||= Requestors::DocumentIdRequestor.new(submission.case_ccms_reference, submission.legal_aid_application.provider.username)
+      def update_document_and_return_response(document, document_id_requestor)
+        document.ccms_document_id = ccms_document_id(document_id_requestor)
+        document.status = :id_obtained
+        document_id_requestor.call
```

Comment:
couldn't you send back `document.ccms_document_id` instead of calling `document_id_requestor.call`? Looks like it's calling here aswell that's why in `ccms_document_id(document_id_requestor)`